src/meta-srv/src/table_meta_alloc.rs

```diff
@@ -58,13 +57,11 @@ impl TableMetadataAllocator for MetaSrvTableMetadataAllocator {
     async fn create(
         &self,
         ctx: &TableMetadataAllocatorContext,
-        raw_table_info: &mut RawTableInfo,
-        partitions: &[Partition],
+        task: &mut CreateTableTask,
     ) -> MetaResult<TableMetadata> {
```

Comment:
I have to disagree with changing `raw_table_info: &mut RawTableInfo, partitions: &[Partition]` to `task: &mut CreateTableTask.`. 

The `&mut CreateTableTask` means all fields of it might be changed. However, this method wouldn't set the `CreateTableExpr`'s `table_id` of the `CreateTableTask`. I prefer letting the developer know the field wouldn't be changed instead of a `&mut CreateTableTask`.